### PR TITLE
Adding acceptance test for Post Location as resource payload validati…

### PR DESCRIPTION
**Pull request by SachinNaikHMCTS**
…on tests

**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: src/acceptanceTest/java/uk/gov/hmcts/futurehearings/snl/acceptance/resources/POSTResourcesByUserPayloadValidationTest.java [hunk 1]
```diff
@@ -249,21 +259,141 @@ public void test_negative_response_for_person_salutation_with_mandatory_elements
                 new SNLVerificationDTO(getHttpSuccessStatus(), null, null, null));
     }
 
-    private void generateResourcesByUserPayloadWithRandomHMCTSId() throws IOException {
-        final String randomID = UUID.randomUUID().toString() + UUID.randomUUID().toString();
-        this.setInputBodyPayload(String.format(TestingUtils.readFileContents(String.format(INPUT_TEMPLATE_FILE_PATH,
-                getInputFileDirectory()) + "/" + getInputPayloadFileName()), randomID));
+    @ParameterizedTest(name = "PersonRoleID Positive Tests Scenario : {0}")
```
Comment (bmj-vsalivendra) [inline src/acceptanceTest/java/uk/gov/hmcts/futurehearings/snl/acceptance/resources/POSTResourcesByUserPayloadValidationTest.java:262]: line262 - Change from positive to negative

File: src/acceptanceTest/java/uk/gov/hmcts/futurehearings/snl/acceptance/resources/POSTResourcesByUserPayloadValidationTest.java [hunk 2]
```diff
@@ -249,21 +259,141 @@ public void test_negative_response_for_person_salutation_with_mandatory_elements
...
+                delegateDTO,
+                getSnlSuccessVerifier(),
+                new SNLVerificationDTO(getHttpSuccessStatus(), null, null, null));
     }
 
-    private void generateResourcesByUserPayloadWithRandomHMCTSIDAndField(final String formatValue) throws IOException {
-        final String randomID = UUID.randomUUID().toString() + UUID.randomUUID().toString();
-        this.setInputBodyPayload(String.format(TestingUtils.readFileContents(String.format(INPUT_TEMPLATE_FILE_PATH,
-                getInputFileDirectory()) + "/" + getInputPayloadFileName()), randomID, formatValue));
+    @ParameterizedTest(name = "PersonVenueID Positive Tests Scenario : {0}")
```
Comment (bmj-vsalivendra) [inline src/acceptanceTest/java/uk/gov/hmcts/futurehearings/snl/acceptance/resources/POSTResourcesByUserPayloadValidationTest.java:293]: line 293: Change from positive to negative

File: src/acceptanceTest/java/uk/gov/hmcts/futurehearings/snl/acceptance/resources/POSTResourcesByUserPayloadValidationTest.java [hunk 3]
```diff
@@ -249,21 +259,141 @@ public void test_negative_response_for_person_salutation_with_mandatory_elements
...
+    @CsvSource(value = {"Empty Space,''","Null Value,NIL","Invalid LOV Value,299","Invalid LOV Value,387"}, nullValues = "NIL")
+    public void test_negative_response_for_person_venue_id_with_mandatory_elements_payload(final String personVenueIdKey,
+                                                                                          final String personVenueIdValue) throws Exception {
+
+        this.setInputPayloadFileName("resources-by-username-mandatory-person-role-id.json");
```
Comment (bmj-vsalivendra) [inline src/acceptanceTest/java/uk/gov/hmcts/futurehearings/snl/acceptance/resources/POSTResourcesByUserPayloadValidationTest.java:298]: line 298 : .json file should be person venu id

File: src/acceptanceTest/java/uk/gov/hmcts/futurehearings/snl/acceptance/resources/POSTResourcesByUserPayloadValidationTest.java [hunk 4]
```diff
@@ -249,21 +259,141 @@ public void test_negative_response_for_person_salutation_with_mandatory_elements
...
+        log.debug("The value of the Delegate Payload : " + delegateDTO.inputPayload());
+        commonDelegate.test_expected_response_for_supplied_header(
+                delegateDTO,
+                getSnlErrorVerifier(),
+                new SNLVerificationDTO(HttpStatus.BAD_REQUEST, null, null, null));
+    }
+
+    @ParameterizedTest(name = "PersonActiveDate Positive Tests Scenario : {0} - {1}")
+    @CsvSource(value = {"Valid Person Active Date Value,2002-10-02"}, nullValues = "NIL")
+    public void test_positive_response_for_person_active_date_with_mandatory_elements_payload(final String personRoleIdKey,
```
Comment (bmj-vsalivendra) [inline src/acceptanceTest/java/uk/gov/hmcts/futurehearings/snl/acceptance/resources/POSTResourcesByUserPayloadValidationTest.java:311]: Line 311- Change parameters from personRoleIdKey and value to personActiveDateKey

**Review by bmj-vsalivendra — COMMENTED** [review]
Duplicate lines are there, may be because of copy and paste. Please check the duplicated lines in all tests . Thanks

File: src/acceptanceTest/java/uk/gov/hmcts/futurehearings/snl/acceptance/resources/POSTResourcesByLocationPayloadValidationTest.java [hunk 5]
```diff
@@ -0,0 +1,736 @@
...
+
+
+@Slf4j
+@SpringBootTest(classes = {Application.class})
+@ActiveProfiles("acceptance")
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+@SelectClasses(POSTResourcesByLocationPayloadValidationTest.class)
+@IncludeTags("Post")
+@Disabled("This test is disabled till the till we have an agreement with S&L over the UAT")
+class POSTResourcesByLocationPayloadValidationTest extends ResourcesPayloadValidationTest {
```
[the remark this answers is not in the document]
Comment (pats-john) [reply src/acceptanceTest/java/uk/gov/hmcts/futurehearings/snl/acceptance/resources/POSTResourcesByLocationPayloadValidationTest.java:44]: Addressed all the Outstanding Comments....

**Comment by pats-john** [comment]
Further review requested on 

1.POST User,
2.PUT User and 
3.GET Sessions....

**Review by bmj-vsalivendra — APPROVED** [review]

**Pull request merged**